### PR TITLE
Fix-hostname

### PR DESCRIPTION
**Pull request by al-rigazzi**
This PR addresses an inconstent internal host name representation in the Dragon backend.

### CODE REVIEW

File: smartsim/_core/launcher/dragon/dragonBackend.py [hunk 1]
```diff
@@ -452,7 +452,7 @@ def create_run_policy(
                 if run_request.policy.gpu_affinity:
                     affinity = dragon_policy.Policy.Affinity.SPECIFIC
                     gpu_affinity = run_request.policy.gpu_affinity
-
+            logger.debug(f"Affinity: {affinity}, {cpu_affinity}, {gpu_affinity}")
```
Comment (mellis13) [inline smartsim/_core/launcher/dragon/dragonBackend.py:455]: If we are going to leave this in, can we clarify the message so we don't rely on remembering the order of the debug output:

```python
logger.debug(f"Affinity: {affinity}")
logger.debug(f"CPU affinity: {cpu_affinity}")
logger.debug(f"GPU affinity: {gpu_affinity}")
```

or do the equivalent in a single line

Comment (al-rigazzi) [reply smartsim/_core/launcher/dragon/dragonBackend.py:455]: Sure, let me do this!

Comment (ankona) [reply smartsim/_core/launcher/dragon/dragonBackend.py:455]: Per meeting w/dragon team, the affinity parameter is also being dropped and is not used. I should removed affinity and leave only cpu/cpu affinity lists behind. I think that will clear up your ordering comment?

Comment (mellis13) [reply smartsim/_core/launcher/dragon/dragonBackend.py:455]: Agreed they are dropping the ``affinity`` parameter.  We can still specify CPU and GPU affinity at the same time (and probably want to know which is which) right?

Comment (al-rigazzi) [reply smartsim/_core/launcher/dragon/dragonBackend.py]: Yeah, we'll update it as soon as the new version is out.

**Review by ankona — APPROVED** [review]
LGTM! Thanks for fixing my bug

**Pull request merged**